### ISSUE TITLE
Bug in Add synonyms method 

### ISSUE THREAD

**Issue by sriramsv** [treethought/flask-assistant#31]
There is a bug in the following code where self.entries is a list and the each entry in the list is a dict, however you are trying to fetch the entry in the list by the value of the entry and not by index.
You need to fetch the dict which has the value of the entry and then extend the synonyms 
```python
def add_entry(self, value, synonyms=[]):
        if self.isEnum:
            entry = {'value': value, 'synonyms': value}
        entry = {'value': value, 'synonyms': synonyms}
        self.entries.append(entry)


    def add_synonyms(self, entry, synonyms):
        self.entries[entry].extend(synonyms)
```

**Comment by robwilkes** [comment]
From what I can see add_synonyms isn't actually called anywhere.

I'm assuming from the description, the actual issue _might_ be that annotations are not resolved into synonyms, and linked to entities, when generating the intents.json file, so once uploaded to Dialogflow (via schema command) none of the synonyms are mapped back to an entity (only the main annotation itself)? 

If yes, I've just submitted a pull request which fixes this in my testing.

**Comment by treethought** [comment]
Should hopefully be resolved with #43. I'll keep this open until it's been tested
